be/src/exec/spill/spiller.hpp

```diff
@@ -22,38 +22,99 @@
 #include "common/logging.h"
 #include "common/status.h"
 #include "exec/spill/common.h"
+#include "exec/spill/serde.h"
 #include "exec/spill/spiller.h"
 #include "util/defer_op.h"
 
-namespace starrocks {
-namespace spill {
+namespace starrocks::spill {
 template <class TaskExecutor, class MemGuard>
-Status Spiller::spill(RuntimeState* state, ChunkPtr chunk, TaskExecutor&& executor, MemGuard&& guard) {
+Status Spiller::spill(RuntimeState* state, const ChunkPtr& chunk, TaskExecutor&& executor, MemGuard&& guard) {
     SCOPED_TIMER(_metrics.spill_timer);
     RETURN_IF_ERROR(_spilled_task_status);
     DCHECK(!chunk->is_empty());
     DCHECK(!is_full());
 
-    if (_mem_table == nullptr) {
-        _mem_table = _acquire_mem_table_from_pool();
-        DCHECK(_mem_table != nullptr);
-    }
-
     COUNTER_UPDATE(_metrics.spill_rows, chunk->num_rows());
     _spilled_append_rows += chunk->num_rows();
     TRACE_SPILL_LOG << "spilled rows:" << chunk->num_rows() << ",cumulative:" << _spilled_append_rows
-                    << ",spiller:" << this << "," << _mem_table.get();
-    RETURN_IF_ERROR(_mem_table->append(std::move(chunk)));
-    if (_mem_table->is_full()) {
-        RETURN_IF_ERROR(flush(state, std::forward<TaskExecutor>(executor), std::forward<MemGuard>(guard)));
+                    << ",spiller:" << this;
+
+    if (_chunk_builder.chunk_schema()->empty()) {
+        _chunk_builder.chunk_schema()->set_schema(chunk);
```

Comment:
can we know the schema of chunk in Operator::prepare stage? I prefer to set it here

Reply:
We can't know the exact schema of the chunk in the prepare phase now (e.g. nullable information)